### DISCUSSION TITLE
Strange Behavior

### DISCUSSION THREAD

**Discussion by scottcantwell** [VsixCommunity/Community.VisualStudio.Toolkit#392] [Q&A] [answered]
I just started to experience this issue if I delete or rename a file in the Solution Explorer while debugging my extensions. Happens right after the confirmation dialog. These message boxes will display.

I thought it might be caused by my code in **OnAfterRenameFiles**, or in  **OnAfterRemoveFiles** but I've commented out the code and it still happens. Doesn't matter what extension I am debugging, it still happens.

VS Version: Microsoft Visual Studio Enterprise 2022 (64-bit) - Current
Version 17.4.0

When deleting a file:

![image](https://user-images.githubusercontent.com/90006729/203209726-3981e3c1-5ae1-425a-b64f-502bdbcf3fca.png)

When renaming  a file:

![image](https://user-images.githubusercontent.com/90006729/203210636-bd825a8d-6149-42b2-9879-95162d639d28.png)

**Comment by scottcantwell** [comment]
Update #1

Not sure if this has anything to do with the problem, but the issue did occur after this came up:

![image](https://user-images.githubusercontent.com/90006729/203218613-7e0938b1-09e5-4c0e-8212-b81d5534a3e7.png)

**Comment by reduckted** [comment, marked as answer]
That would be caused by these lines in the demo extension.

https://github.com/VsixCommunity/Community.VisualStudio.Toolkit/blob/20209526189866985f693bf2b1a4fa04a126e089/demo/VSSDK.TestExtension/TestExtensionPackage.cs#L88-L98

Presumably you've run up the demo extension at some point. Once an extension is deployed to the experimental instance, it stays there and is used by Visual Studio even if you aren't debugging that extension, which can be a bit annoying, especially if there are bugs in it. 😆 

If you reset the experimental instance, then it will effectively uninstall any extensions that you've debugged.

**Reply by scottcantwell** [reply]
Thanks. 

I saw that the other day and was going to post an update. 

Scott
